src/flamegraph/mod.rs

```diff
@@ -325,19 +333,51 @@ impl Rectangle {
 ///
 /// [differential flame graph]: http://www.brendangregg.com/blog/2014-11-09/differential-flame-graphs.html
 #[allow(clippy::cyclomatic_complexity)]
-pub fn from_sorted_lines<'a, I, W>(opt: &mut Options, lines: I, writer: W) -> quick_xml::Result<()>
+pub fn from_lines<'a, I, W>(opt: &mut Options, lines: I, writer: W) -> quick_xml::Result<()>
 where
     I: IntoIterator<Item = &'a str>,
     W: Write,
 {
-    let (bgcolor1, bgcolor2) = color::bgcolor_for(opt.bgcolors, opt.colors);
+    let mut reversed = StrStack::new();
+    let (mut frames, time, ignored, delta_max) = if !opt.no_sort && !opt.reverse_stack_order {
```

Comment:
Can we just do `let lines = if ...` and then place the call to `merge::frames` on its own line after? I think that'd be more readable.

Reply:
But in the `else` we don't create a `Vec`, but we do in the other two blocks. So the if and else would have incompatible types.

Reply:
Ohhh, I missed that, you are completely right of course. Never mind me :angel: